src/core/java/com/enderio/core/common/blockentity/EnderBlockEntity.java

```diff
@@ -132,10 +165,11 @@ public <T> void clientUpdateSlot(@Nullable NetworkDataSlot<T> slot, T value) {
         }
 
         if (dataSlots.contains(slot)) {
-            CompoundTag updateData = new CompoundTag();
-            updateData.putInt("Index", dataSlots.indexOf(slot));
-            updateData.put("Data", slot.serializeValueNBT(value));
-            CoreNetwork.sendToServer(new C2SDataSlotChange(getBlockPos(), updateData));
+            FriendlyByteBuf buf = new FriendlyByteBuf(Unpooled.buffer());
+            buf.writeInt(dataSlots.indexOf(slot));
+            slot.toBuffer(buf, value);
+            CoreNetwork.sendToServer(new C2SDataSlotChange(getBlockPos(), buf));
+            level.sendBlockUpdated(getBlockPos(), getBlockState(), getBlockState(), Block.UPDATE_NEIGHBORS);
```

Comment:
If we want we could reduce this by only updating when needed, like the world and GUI flags we had before.

Reply:
yes, but later